### PR TITLE
added solution to task 2 "equals & hashcode"

### PR DESCRIPTION
**Pull request by SkaterWeather**

### CODE REVIEW

File: src/main/java/core/basesyntax/Rectangle.java [hunk 1]
```diff
@@ -27,4 +27,23 @@ public Integer getLength() {
...
+        }
+        return (object instanceof Rectangle)
+                && ((Rectangle) object).getWidth().equals(this.width)
+                && ((Rectangle) object).getLength().equals(this.length)
+                && ((Rectangle) object).getColor().equals(this.color);
+    }
+
+    @Override
+    public int hashCode() {
+        return width.hashCode() + length.hashCode() + color.hashCode();
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Rectangle.java]: @SkaterWeather let's use prime numbers

File: src/main/java/core/basesyntax/Rectangle.java [hunk 2]
```diff
@@ -27,4 +27,23 @@ public Integer getLength() {
...
+    @Override
+    public boolean equals(Object object) {
+        if (object == null) {
+            return false;
+        }
+        if (object == this) {
+            return true;
+        }
+        return (object instanceof Rectangle)
+                && ((Rectangle) object).getWidth().equals(this.width)
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Rectangle.java]: Let's create one object and not cast every time

**Review by boroda4436 — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Rectangle.java [hunk 3]
```diff
@@ -27,4 +27,23 @@ public Integer getLength() {
...
+
+    @Override
+    public boolean equals(Object object) {
+        if (object == this) {
+            return true;
+        }
+        if (object == null) {
+            return false;
+        }
+        Rectangle rectangle = (Rectangle) object;
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Rectangle.java]: @SkaterWeather What is object will be Plane not Rectangle?

Comment (SkaterWeather) [reply src/main/java/core/basesyntax/Rectangle.java]: I'am accidentally removed checking "inctanceof", sorry

**Pull request closed without merging**